tests/ui/or_fun_call.fixed

```diff
@@ -176,4 +176,52 @@ mod issue6675 {
     }
 }
 
+mod issue8239 {
+    unsafe fn more_than_max_suggestion_highest_lines_0() {
+        let frames = Vec::new();
+        frames
+            .iter()
+            .map(|f: &String| f.to_lowercase())
+            .reduce(|mut acc, f| {
+                acc.push_str(&f);
+                acc
+            })
+            .unwrap_or_default();
+    }
+
+    unsafe fn more_to_max_suggestion_highest_lines_1() {
+        let frames = Vec::new();
+        let iter = frames.iter();
+        iter.map(|f: &String| f.to_lowercase())
+            .reduce(|mut acc, f| {
+                let _ = "";
+                let _ = "";
+                acc.push_str(&f);
+                acc
+            })
+            .unwrap_or_default();
+    }
+
+    unsafe fn equal_to_max_suggestion_highest_lines() {
+        let frames = Vec::new();
+        let iter = frames.iter();
+        iter.map(|f: &String| f.to_lowercase())
+            .reduce(|mut acc, f| {
+                let _ = "";
+                acc.push_str(&f);
+                acc
+            }).unwrap_or_default();
+    }
+
+    unsafe fn less_than_max_suggestion_highest_lines() {
```

Comment:
Is there a reason why all of these test functions are unsafe? :) There is nothing wrong with it, I'm just curious.

Reply:
> This version looks good to me, one optional NIT and one question.
> 
> BTW, all your commits are marked as _Unverified_. This is not a problem, but I wanted to mention it in case this was unintended 🙃

Should I upload my public GPG key to GitHub to verify my commits ?

Reply:
> Is there a reason why all of these test functions are unsafe? :) There is nothing wrong with it, I'm just curious.

I made them unsafe without any reason - will fix it in next commit.

Reply:
> Should I upload my public GPG key to GitHub to verify my commits ?

Either is fine for us. We can merge this as it is. I just like to point these things out :upside_down_face:

Reply:
> > Should I upload my public GPG key to GitHub to verify my commits ?
> 
> Either is fine for us. We can merge this as it is. I just like to point these things out 🙃

Ok uploaded public GPG key but still Unverified in here. Probably next PR will be Verified :) Will C :)

Reply:
Alright :+1: